crates/core/src/rpc/minimal.rs

```diff
@@ -752,6 +780,100 @@ impl Minimal for SurfpoolMinimalRpc {
 mod tests {
     use super::*;
     use crate::tests::helpers::TestSetup;
+    use solana_commitment_config::{CommitmentConfig, CommitmentLevel};
+    use solana_client::rpc_config::RpcContextConfig;
+    use solana_epoch_info::EpochInfo;
+
+    #[test]
+    fn test_get_block_height_basic() {
+        let setup = TestSetup::new(SurfpoolMinimalRpc);
+        
+        let result = setup.rpc.get_block_height(Some(setup.context.clone()), None);
+        assert!(result.is_ok(), "get_block_height should not fail");
+
+        let block_height = result.unwrap();
+
+        println!("Block height returned: {:?}", block_height);
+    }
+
+    #[test]
+    fn test_get_block_height_matches_svm_state() {
+        let setup = TestSetup::new(SurfpoolMinimalRpc);
+
+        let result = setup.rpc.get_block_height(Some(setup.context.clone()), None);
+        assert!(result.is_ok());
+
+        let block_height = result.unwrap();
+        let svm_block_height = setup.context.surfnet_svm.blocking_read().latest_epoch_info.block_height;
+
+        assert_eq!(block_height, svm_block_height,
+                   "Implementation should return the same block height as SVM state");
+    }
+
+    #[test]
+    fn test_get_block_height_with_processed_commitment() {
+        let setup = TestSetup::new(SurfpoolMinimalRpc);
+
+        let config = Some(RpcContextConfig {
+            commitment: Some(CommitmentConfig {
+                commitment: CommitmentLevel::Processed,
+            }),
+            min_context_slot: None,
+        });
+
+        let result = setup.rpc.get_block_height(Some(setup.context.clone()), config);
+        assert!(result.is_ok());
+
+        let block_height = result.unwrap();
+        let expected_height = setup.context.surfnet_svm.blocking_read().latest_epoch_info.block_height;
+        assert_eq!(block_height, expected_height);
+    }
+
+    #[test]
+    fn test_get_block_height_with_valid_min_context_slot() {
```

Comment:
Good test! Can we also get one for the failure case - asserting the RPC returns an error if the min_context_slot is beyond the current slot?